lib/db_connection/ownership/proxy.ex

```diff
@@ -157,32 +165,60 @@ defmodule DBConnection.Ownership.Proxy do
 
   # If it is down but it has no client, checkin
   defp down(reason, %{client: nil} = state) do
-    pool_done(reason, state, fn pool_ref, _ ->
-      Holder.checkin(pool_ref)
-    end)
+    pool_checkin(reason, state)
   end
 
-  # If it is down but it has a client, disconnect
+  # If it is down but it has a client, disconnect but do not log
   defp down(reason, state) do
     err = DBConnection.ConnectionError.exception(reason)
     pool_disconnect(err, state)
   end
 
+  defp log_and_disconnect(reason, state) do
+    err = DBConnection.ConnectionError.exception(reason)
+
+    _ = Logger.error(fn() ->
+      ["Ownership connection (", inspect(self()), ") disconnected: "
```

Comment:
Will this be too noisy as the connection process will log to? Is this in case pre_checkin has an issue (crashes) and we want to see this first?

Reply:
We were not logging anything before this change. I found this out while writing the new tests. So it is not related to pre_checkin but it was necessary (it was even causing failures in Ecto suite).

We log in the following cases:

1. owner exit while client is running
2. owner timeout
3. client timeout

I think it is important to log in those cases because they will lead to cascading changes and we may lose the root cause when those happen. For which case above are you worried about excessive messages? All three of them?

Note there are cases we don't log: client exit and owner exit without client.

Reply:
Actually, you are correct, we are logging twice in db connection tests. However, we don't get the logs in ecto tests. I am investigating.

Reply:
I will send a PR and continue the discussion there.